dsmr_parser/clients/protocol.py

```diff
@@ -99,7 +102,7 @@ def connection_made(self, transport):
 
     def data_received(self, data):
         """Add incoming data to buffer."""
-        data = data.decode('ascii')
+        data = data.decode('ascii', errors='ignore')
```

Comment:
Yes, thought about it as well, but I saw a few identical or similar issues reported that this would fix and other forks solved it this way as well.

How would you imagine this would break things?

This is a quite simple fix. 

Maybe additional checks further downstream could drop the telegram if needed.

Reply:
I don't know honestly.

But here we have one meter, that is adding a trailing  \xFF. 
The code is changed to ignore any decode error on the complete stream.
So suddenly it is allowing way more probably really erroneous (while un-decodeable) cases.
That does not sound like a good way of handling this.

What about the following solution; we add a new decode_ascii_telegram method, which:
1. removes all trailing \xFF `data.rstrip(b'\xff')`
2. invokes the `data.decode('ascii')`

and invoke this in all places where currently `data.decode('ascii')` is used.

Reply:
See my other comment on the garbage added by the COM-1 boxes.

I would say there is sufficient downstream checks with _find_telegrams to ignore the garbage and with the CRC-checks to check the telegram integrity (if CRC is available). Futhermore the Regexs should also fail in cases where lines are incomplete (w/ or w/o CRC).

Unfortunatly \xff are not the last chars, but it's terminated by CR LF. Thus rstrip won't work.

I'm no expert here, but if I understand decode correctly it's just failing when it can't map a bytecode to target charset. It wouldn't catch any other "errors". So generally the "really erroneous" cases are quite limited, I would think. errors="replace" could be an option to "see" these cases.

Using "latin1" (8-bit) instead "ascii" (7-bit) works fine. Any concerns?